aiosparql/test_utils.py

```diff
@@ -1,16 +1,81 @@
-from aiohttp.test_utils import AioHTTPTestCase, TestClient, unittest_run_loop
+from aiohttp.test_utils import (
+    AioHTTPTestCase, BaseTestServer, TestServer, unittest_run_loop)
 from aiosparql.client import SPARQLClient
 from aiosparql.syntax import IRI
-import asyncio
 
 
 __all__ = ['unittest_run_loop', 'AioSPARQLTestCase', 'TestSPARQLClient']
 
 
-class TestSPARQLClient(TestClient, SPARQLClient):
-    def __init__(self, app, **kwargs):
-        TestClient.__init__(self, app, loop=kwargs.get('loop'))
-        SPARQLClient.__init__(self, **kwargs)
+class TestSPARQLClient:
+    def __init__(self, server, *, cookie_jar=None, loop=None, **kwargs):
+        if not isinstance(server, BaseTestServer):
+            raise TypeError("server must be web.Application TestServer "
+                            "instance, found type: %r" % type(server))
+        self._server = server
+        self._loop = loop
+        self._client_kwargs = kwargs
+        self._session = None
+        self._closed = False
+
+    async def start_server(self):
+        await self._server.start_server(loop=self._loop)
+        kwargs = dict(self._client_kwargs)
+        if kwargs.get('endpoint'):
+            kwargs['endpoint'] = self.make_url(kwargs['endpoint'])
+        if kwargs.get('update_endpoint'):
+            kwargs['update_endpoint'] = \
+                self.make_url(kwargs['update_endpoint'])
+        self._session = SPARQLClient(loop=self._loop,
+                                     **kwargs)
```

Comment:
I moved all this code from `__init__` because I need the server to be started before initializing the SPARQLClient.

The big difference between the SPARQLClient and ClientSession is that there is no `url` argument. The URL is define at object instantiation and the user just create a new SPARQLClient whenever they want to query a different URL.

This is by design.